archer_utils.py

```diff
@@ -683,32 +701,255 @@ def update_record(self, app, contentId, fieldId, value, doit=True):
+        fields = []
+
+        for field, value in list(data.items()):
+            fid = ''
+            try:
+                fid = int(field)
+            except (ValueError, TypeError):
+                fid = self.get_fieldId_for_app_and_name(app, field)
+            try:
+                fid = int(fid)
+                W('fieldId is integer, using as-is: {}'.format(fid))
+            except (ValueError, TypeError):
+                newId = self.get_fieldId_for_app_and_name(app, field)
+                W('Got fieldId from app_and_name: {}'.format(newId))
+                if newId is None:
+                    raise Exception("Can't resolve field: {}".format(field))
+                elif doit:
+                    raise Exception("Can't update without content_id")
+                fid = newId
+
+            fieldType = int(self.get_type_of_field(fid))
+            W('Got fieldType: {}'.format(fieldType))
+            levelId = self.get_level_of_field(fid)
+            W('Got levelId: {}'.format(levelId))
+            value = self.get_valuesetvalue_of_field(fid, value)
+            moduleId = self.get_moduleid(app)
+            field = {
+                'id': fid,
+                'type': fieldType,
+                'value': value
+            }
+            W('Updating to value: {}'.format(value))
+            W('Updating to id: {}'.format(fid))
+            W('Updating to type: {}'.format(fieldType))
+            fields.append(field)
+        data = self.asoap.update_record(contentId, moduleId, fields)
+        return bool(data)
+
+    def get_report_by_id(self, guid, max_count, max_pages):
+        """Returns the report with the given guid."""
+
+        # Initialize result dictionary
+        result_dict = {}
+        result_dict['status'] = 'failed'
+        result_dict['message'] = 'Failed - default message'
+        result_dict['page_count'] = 0
+        result_dict['records'] = []
+
+        # Initialize the current count of records
+        total_count = 0
+
+        # Try to loop through report pages until no records are returned, max pages reached,
+        # or max number of record results reached
         try:
-            if self.sessionToken:
-                result = self.asoap.terminate_session(self.sessionToken)
-            else:
-                result = self.asoap.terminate_session(self.get_token())
+
+            for page_number in range(1, max_pages + 1):
+
+                # Try to get current report page
+                try:
+                    data_dict = self.asoap.get_report(guid, page_number)
+                    if data_dict['status'] != 'success':
+                        result_dict['message'] = data_dict['result']
+                        return result_dict
+                    data = data_dict['result']
+
+                except Exception as e:
+                    result_dict['message'] = 'Failed to get page {} of report. Check input parameters are valid. e = {}'.format(page_number, e)
```

Comment:
Better to use the word error instead of just e here since it is the message to the user

Reply:
Same with other occurrences